src/dialogs/edittagdialog.cpp

```diff
@@ -1411,6 +1415,23 @@ void EditTagDialog::FetchTagSongChosen(const Song &original_song, const Song &ne
 
 }
 
+void EditTagDialog::FetchLyrics() {
+
+  Song song = data_[ui_->song_list->selectionModel()->selectedIndexes().first().row()].original_;
+  lyrics_fetcher_->Clear();
+  lyrics_fetcher_->Search(song.effective_albumartist(), song.artist(), song.album(), song.title());
```

Comment:
You need to save the ID to a member variable, so if the user selects another song while the lyrics are fetched, it does not use that lyrics.